src/cryptonote_basic/hardfork.cpp

```diff
@@ -379,18 +379,21 @@ uint8_t HardFork::get_ideal_version(uint64_t height) const
 
 uint64_t HardFork::get_earliest_ideal_height_for_version(uint8_t version) const
 {
-  for (unsigned int n = heights.size() - 1; n > 0; --n) {
-    if (heights[n].version <= version)
-      return heights[n].height;
+  uint64_t height = std::numeric_limits<uint64_t>::max();
+  for (int n = heights.size() - 1; n >= 0; --n) {
```

Comment:
probably better to rewrite the cycle this way:
```
for (auto rit = heights.rbegin(); rit != heights.rend(); ++rit) {
  if (rit->version >= version) {
    height = rit->height;
  } else {
    break;
  }
}

it's less error prone and looks more natural

Reply:
Agreed, thanks.